backend/lib/truecallerUserLib.js

```diff
@@ -80,3 +80,35 @@ module.exports.getRecordByNumber = async (req, res) => {
     res.status(500).json(err);
   }
 }
+
+//function to find a document and update it 
+module.exports.findAndUpdate = async (req, res) => {
+  try {
+    let data = req.body;
+    let filter = { phone: data.oldphone }
+    let duplicateRecord = false
+    if (data.oldphone != data.phone) {
+      //checking if updated phone number already exits in DB
+      duplicateRecord = await asyncDbLib.getOneDocumentByFilter(truecallerUserModel, { phone: req?.body?.phone })
+      logger.debug("duplicate Record is ", duplicateRecord)
+    }
+    if (duplicateRecord) {
+      res.status(409).json(duplicateRecord);
+    }
+    else {
+      try {
+        const result = await asyncDbLib.findOneDocumentByFilterAndUpdate(truecallerUserModel, filter, data)
+        logger.debug("result is ", result)
```

Comment:
won't it take automatic space while printing?

Reply:
Its user wish 
Not a standard